lib/Mojo/mysql.pm

```diff
@@ -73,7 +73,11 @@ sub from_string {
   return $self->dsn($dsn);
 }
 
-sub new { @_ == 2 ? shift->SUPER::new->from_string(@_) : shift->SUPER::new(@_) }
+sub new {
+    @_ > 2 || ref $_[-1] eq 'HASH'
+        ? shift->SUPER::new(@_)
+        : shift->SUPER::new->from_string(@_)
```

Comment:
Please tidy both files, if you want `git blame` to reference you later on 😄

Reply:
Done!